### ISSUE TITLE
Stack overflow trying to resolve certain TypeSignature and MemberReference

### ISSUE THREAD

**Issue by sgkoishi** [Washi1337/AsmResolver#381]
### AsmResolver Version

5.0.0-beta.2

### .NET Version

.NET 6.0

### Operating System

Windows

### Describe the Bug

Stack overflow trying to resolve certain `TypeSignature` and `MemberReference`.

### How To Reproduce

```csharp
foreach (var type in asm.ManifestModule.GetAllTypes())
{
    foreach (var method in type.Methods)
    {
        foreach (var target in method.Parameters)
        {
            // target.ParameterType.Resolve();
        }
        if (method.CilMethodBody == null)
        {
            continue;
        }
        foreach (var instr in method.CilMethodBody.Instructions)
        {
            if (instr.Operand is MemberReference mr)
            {
                // mr.Resolve();
            }
        }
    }
}
```

### Expected Behavior

Return resolved `TypeDefinition` or null.

### Actual Behavior

For `MemberReference`
```
Stack overflow.
Repeat 5065 times:
--------------------------------
   at System.Collections.Generic.Dictionary`2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].FindValue(System.__Canon)
   at AsmResolver.DotNet.AssemblyResolverBase.Resolve(AsmResolver.DotNet.AssemblyDescriptor)
   at AsmResolver.DotNet.AssemblyReference.GetPublicKeyToken()
   at AsmResolver.DotNet.Signatures.SignatureComparer.GetHashCode(AsmResolver.DotNet.AssemblyDescriptor)
--------------------------------
   at System.Collections.Generic.Dictionary`2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].FindValue(System.__Canon)
   at AsmResolver.DotNet.AssemblyResolverBase.Resolve(AsmResolver.DotNet.AssemblyDescriptor)
   at AsmResolver.DotNet.DefaultMetadataResolver+TypeResolution.ResolveTypeReference(AsmResolver.DotNet.TypeReference)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveTypeReference(AsmResolver.DotNet.TypeReference)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveTypeSignature(AsmResolver.DotNet.Signatures.Types.TypeSignature)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveType(AsmResolver.DotNet.ITypeDescriptor)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveType(AsmResolver.DotNet.ITypeDescriptor)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveMethod(AsmResolver.DotNet.IMethodDescriptor)
```

For `TypeSignature`
```
Stack overflow.
Repeat 5065 times:
--------------------------------
   at System.Collections.Generic.Dictionary`2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].FindValue(System.__Canon)
   at AsmResolver.DotNet.AssemblyResolverBase.Resolve(AsmResolver.DotNet.AssemblyDescriptor)
   at AsmResolver.DotNet.AssemblyReference.GetPublicKeyToken()
   at AsmResolver.DotNet.Signatures.SignatureComparer.GetHashCode(AsmResolver.DotNet.AssemblyDescriptor)
--------------------------------
   at System.Collections.Generic.Dictionary`2[[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].FindValue(System.__Canon)
   at AsmResolver.DotNet.AssemblyResolverBase.Resolve(AsmResolver.DotNet.AssemblyDescriptor)
   at AsmResolver.DotNet.DefaultMetadataResolver+TypeResolution.ResolveTypeReference(AsmResolver.DotNet.TypeReference)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveTypeReference(AsmResolver.DotNet.TypeReference)
   at AsmResolver.DotNet.DefaultMetadataResolver.ResolveType(AsmResolver.DotNet.ITypeDescriptor)
```

### Additional Context

[Target assembly on NuGet: `OTAPI.Upcoming` v3.1.20](https://www.nuget.org/packages/OTAPI.Upcoming/3.1.20)

**Comment by Washi1337** [comment]
Potential fix is in #382. Could you maybe verify if this indeed solves your problem?

**Comment by sgkoishi** [comment]
Yes, it's fixed in #382 bf4ed23

**Comment by Washi1337** [comment]
Nice, this fix will be soon be present on nuget together with the full release of 5.0.

Thanks for reporting once again :)
